pkg/controller/operators/catalog/operator.go

```diff
@@ -1167,6 +1168,9 @@ func (o *Operator) ensureInstallPlan(logger *logrus.Entry, namespace string, gen
 		return nil, err
 	}
 
+	o.muInstallPlan.Lock()
+	defer o.muInstallPlan.Unlock()
```

Comment:
Coming back to this after this has been siting for a while: can we add an explanation as to why this might be needed?

Reply:
Yes, I think so.  I can add the explanation in a comment here.